src/swap/action.rs

```diff
@@ -141,16 +174,23 @@ where
     }
 }
 
+/// Result of doing a conditional protocol action.
+///
+/// If the action was done successfully we `Continue` and obtain the
+/// event `E`. Otherwise we `Abort`.
```

Comment:
> I thought we aborted as soon as the expiry is reached even if the action is in progress?

"Otherwise" does cover the situation where the action is still in progress. I can clarify that in the docstring.

Reply:
I actually think `Result` might suffice as you suggest. Originally this enum had more variants. The error and abort returned by this function are always handled in the same way by `hbit_herc20` and `herc20_hbit` anyway.